src/components/course/enrollment/tests/hooks.test.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+
+import { CourseContextProvider } from '../../CourseContextProvider';
+import { UserSubsidyContext } from '../../../enterprise-user-subsidy';
+
+import { useEnrollData, useSubsidyData } from '../hooks';
+
+const BASE_COURSE_STATE = {
+  activeCourseRun: {
+    firstEnrollablePaidSeatPrice: 7.50,
+    key: 'course_key',
+    start: '2020-02-12T10:00:00Z',
+    isEnrollable: false,
+  },
+  userSubsidyApplicableToCourse: null,
+  course: {},
+  userEnrollments: [],
+  userEntitlements: [],
+  catalog: [],
+};
+
+describe('useEnrollData tests', () => {
+  test('correct extraction of enroll fields', () => {
+    const userEnrollment = undefined;
+    const expected = {
+      isEnrollable: false,
+      isUserEnrolled: false, // since no userEnrollments are passed
```

Comment:
I'd like to see a few more tests to determine that it correctly extracts this if there are userEnrollments and/or if the course is started, etc.

Reply:
sure! this is part of the functionality is in scope to do this